dev/storage_base.h

```diff
@@ -728,6 +728,20 @@ namespace sqlite_orm {
                 }
             }
 
+            // migration internal API
+            void start_migration() {
+                this->pragma.foreign_keys(false);
```

Comment:
there is no need to disable FK. We've told about already. You can migrate data using raw SQLite without disabling FK then it must work the same way in `sqlite_orm`

Reply:
Table Emp is the target of Department and thus we cannot drop the Emp with FK on

Reply:
don't drop it, rename it and create a new one

Reply:
> there is no need to disable FK. 
look at the schema:

```C++
make_table("Dept",
                       make_column("deptno", &Department::m_deptno, primary_key(), autoincrement()),
                       make_column("deptname", &Department::m_deptname),
                       make_column("loc", &Department::m_loc),
                       make_column("mgr", &Department::m_manager),
                       foreign_key(&Department::m_manager).references(&Employee::m_empno)),
```

with this FK we cannot drop Employee table with FK ON!!
Why do you insist that this is possible?

Reply:
please pay attention to my ideas - I am going to address your objections after my swimming 
please have an open mind
the goal with sync_schema is to not lose data while syncronizing , it should always be called inside try/block so user can see what existing data is not allowing the synchronization and correct it!

Reply:
> Why do you insist that this is possible?

Cause it is possible.

> with this FK we cannot drop Employee table with FK ON!!

don't drop it, rename it and create a new one.

Reply:
> the goal with sync_schema is to not lose data while synchronizing 

why do you think so? One of scenarios of `sync_schema` is losing data, it is noted in a comment. `sync_schema` means `please synchronize schema in db just like I wrote in C++ inside make_storage call`. If it will not synchronize by artificial reasons then it means that function doesn't work. If someone is afraid to lose data there is an easy way to look into the future and to know what will happen if he/she will call `sync_schema` - to call `sync_schema_simulate` and check the result. What you are doing in this PR can already be achieved with different API. So please don't affect current `sync_schema` success results (BTW current unit tests failure tell about it). If you want to make life better for people who loses data during `sync_schema` calls then you need to
1) keep current `sync_schema` success results, skip changing algo to 'oops data is lost so no sync, here is an exception for even if you don't care about data';
2) add some API from side which will allow to handle cases when data is removed. You can place it under `if (preserve)` flag inside `sync_schema` cause `preserve` argument is designed for your case especially or make a new function

Reply:
> > Why do you insist that this is possible?
> 
> Cause it is possible.
> 
> > with this FK we cannot drop Employee table with FK ON!!
> 
> don't drop it, rename it and create a new one.

I thought renaming was also prohibited if FK was ON!!
Now I know different! Will try to use this in backup_table

Reply:
the best way of adding functionality to `sqlite_orm` is to make it using raw SQLite first